account_billing/tests/test_account_billing.py

```diff
@@ -0,0 +1,134 @@
+        cls.invoice_line_model = cls.env['account.invoice.line']
+        cls.billing_model = cls.env['account.billing']
+        cls.register_payments_model = cls.env['account.register.payments']
+
+        cls.partner_agrolait = cls.env.ref("base.res_partner_2")
+        cls.partner_china_exp = cls.env.ref("base.res_partner_3")
+        cls.product = cls.env.ref("product.product_product_4")
+        cls.currency_chf_id = cls.env.ref("base.CHF").id
+        cls.currency_usd_id = cls.env.ref("base.USD").id
+        cls.currency_eur_id = cls.env.ref("base.EUR").id
+
+        cls.account_receivable = cls.env['account.account'].search([
+            ('user_type_id', '=',
+             cls.env.ref('account.data_account_type_receivable').id)], limit=1)
+        cls.account_revenue = cls.env['account.account'].search([
+            ('user_type_id', '=',
+             cls.env.ref('account.data_account_type_revenue').id)], limit=1)
+        cls.payment_method_manual_in = cls.env.ref(
+            "account.account_payment_method_manual_in")
+        cls.journal_bank = cls.env['account.journal'].create(
+            {'name': 'Bank', 'type': 'bank', 'code': 'BNK67'})
+
+        cls.inv_1 = cls.create_invoice(
+            cls, amount=100, currency_id=cls.currency_eur_id,
+            partner=cls.partner_agrolait.id)
+        cls.inv_2 = cls.create_invoice(
+            cls, amount=200, currency_id=cls.currency_eur_id,
+            partner=cls.partner_agrolait.id)
+        cls.inv_3 = cls.create_invoice(
+            cls, amount=300, currency_id=cls.currency_usd_id,
+            partner=cls.partner_agrolait.id)
+        cls.inv_4 = cls.create_invoice(
+            cls, amount=400, currency_id=cls.currency_eur_id,
+            partner=cls.partner_china_exp.id)
+
+    def create_invoice(self, amount=None, type='out_invoice',
+                       currency_id=None, partner=None, account_id=None):
+        """ Returns an open invoice """
+        invoice = self.invoice_model.create({
+            'partner_id': partner or self.partner_agrolait.id,
+            'currency_id': currency_id or self.currency_eur_id,
+            'name': type,
+            'account_id': account_id or self.account_receivable.id,
+            'type': type
+        })
+        self.invoice_line_model.create({
+            'product_id': self.product.id,
+            'quantity': 1,
+            'price_unit': amount,
+            'invoice_id': invoice.id,
+            'name': 'something',
+            'account_id': self.account_revenue.id,
+        })
+        invoice.action_invoice_open()
+        return invoice
+
+    def create_payment(self, ctx):
+        register_payments = \
+            self.register_payments_model.with_context(ctx).create({
+                'journal_id': self.journal_bank.id,
+                'payment_method_id': self.payment_method_manual_in.id
+            })
+        return register_payments.create_payments()
+
+    def test_invoice_partner(self):
```

Comment:
On all test, you can order it by naming it. I.e., test_1_xxx, test_2_yyy
It is run by order.